src/main/resources/io/viash/platforms/nextflow/WorkflowHelper.nf

```diff
@@ -603,10 +606,467 @@ def paramsToList(params, config) {
+}
+
+/**
+ * Split parameters for arguments that accept multiple values using their seperator
+ *
+ * @param paramList A Map containing parameters to split.
+ * @param multiArgumentsSettings A List of the viash configuration argument entries 
+ *                               that have the property 'multiple: true'
+ *
+ * @return A Map of parameters where the parameter values have been split into a list using
+ *         their seperator.
+ */
+Map<String, Object> _splitParams(Map<String, Object> parValues, List<Map> multiArgumentsSettings){
+  def parsedParamValues = parValues.collectEntries { parName, parValue ->
+    parameterSettings = multiArgumentsSettings.find({it.plainName == parName})
+    if (parameterSettings) { // Check if parameter can accept multiple values
+      if (parValue instanceof List) {
+          parValue = parValue.collect{it instanceof String ? it.split(parameterSettings.multiple_sep) : it }
+      } else if (parValue instanceof String) {
+          parValue = parValue.split(parameterSettings.multiple_sep)
+      } else if (parValue == null) {
+          parValue = []
+      } else {
+          parValue = [ parValue ]
+      }
+      parValue = parValue.flatten()
+    }
+    // For all parameters check if multiple values are only passed for
+    // arguments that allow it
+    if (parValue instanceof Collection && !parameterSettings ) {
+      assert parValue.size() == 1 : 
+      "Error: argument ${parName} has too many values.\n" +
+      "  Expected amount: 1. Found: ${parValue.size()}"
+      parValue = parValue[0]
+    }
+    [parName, parValue]
+  }
+  return parsedParamValues
+}
+
+/**
+ * Check if the ids are unique across parameter sets
+ *
+ * @param parameterSets a list of parameter sets.
+ */
+private void _checkUniqueIds(List<Tuple2<String, Map<String, Object>>> parameterSets) {
+  def ppIds = parameterSets.collect{it[0]}
+  assert ppIds.size() == ppIds.unique().size() : "All argument sets should have unique ids. Detected ids: $ppIds"
+}
+
+/**
+ * Resolve the file paths in the parameters relative to given path
+ *
+ * @param paramList A Map containing parameters to process.
+ *                  This function assumes that files are still of type String.
+ * @param inputFileSetttings A Map of the viash configuration argument entries 
+ *                           that are considered input files (i.e. with 'direction: input'
+ *                           and 'type: file').
+ * @param relativeTo path of a file to resolve the parameters values to.
+ *
+ * @return A map of parameters where the location of the input file parameters have been resolved
+ *         resolved relatively to the provided path.
+ */
+private Map<String, Object> _resolvePathsRelativeTo(Map paramList, 
+  List<Map<String, String>> inputFileSetttings, String relativeTo) {
+  paramList.collectEntries { parName, parValue ->
+    isInputFile = inputFileSetttings.find({it.plainName == parName})
+    if (isInputFile) {
```

Comment:
Same comment as before. I think its much more readable to pass all arguments to this function than to check whether its in a subset of input arguments.

Proposed alternative:

```groovy
private Map<String, Object> _resolvePathsRelativeTo(
  Map paramList,
  List<Map<String, Object>> arguments,
  String relativeTo
) {
  paramList.collectEntries { parName, parValue ->
    arg = arguments.find{it.plainName == parName}
    if (arg && arg.type == "file" && arg.direction == "input") {
      if (parValue instanceof Collection) {
        parValue = parValue.collect({path -> 
          path !instanceof String ? path : file(getChild(relativeTo, path))
        })
      } else {
        parValue = parValue !instanceof String ? path : file(getChild(relativeTo, parValue))
      }
    }
    [parName, parValue]
  }
}
```

Reply:
Adjusted